### PR TITLE
Feature/fix lookup.country and lookup.asn #54

### PR DESCRIPTION
**Pull request by kidmose**
Fixed error in `lookup.{country|asn}` that was caused by MaxMind closing their public download URL for the two databases. 
Implemented simple authentication with a license key in the URL GET parameters.
Added per method test cases for `lookup.util`.

### CODE REVIEW

**Comment by kidmose** [comment]
Forgot to mention explicitly that this fixes  #54

**Comment by mrturkmen06** [comment]
Hi Egon, 
I have just looked at it however I could not entirely understand why you have provided MAXMIND_LICENSE_KEY, when I check the test error, it is complaining about same error as before.
 
``` _______________________ RetrieveTestCase.test_get_asn _________________________

self = <richkit.test.test_retrieve.RetrieveTestCase testMethod=test_get_asn>

    def test_get_asn(self):
        for k, v in self.test_urls.items():
            instance = URLVoid(k)
>           assert instance.get_asn() == v["ASN"]
E           AssertionError: assert 'AS15169 GOOGLE' == 'AS15169 Google LLC'
E             - AS15169 GOOGLE
E             + AS15169 Google LLC

richkit/test/test_retrieve.py:94: AssertionError 

```

What does this pull request fix, I could not entirely understand, if you do not mind, could you explain it a little bit ? That I can understand better.

File: richkit/lookup/util.py [hunk 1]
```diff
@@ -16,10 +32,15 @@ class MaxMind_CC_DB(object):
...
-    MASTERFILE = temp_directory + "/country.tar.gz"
-
-
+    MASTERURL = ( # https://dev.maxmind.com/geoip/geoipupdate/#Direct_Downloads
+        "https://download.maxmind.com/app/geoip_download?"
+        "edition_id=GeoLite2-Country&"
+        "license_key={license_key}&"
+        "suffix=tar.gz"
+    ).format(
+        license_key=os.environ['MAXMIND_LICENSE_KEY'],
```
Comment (kidmose) [inline richkit/lookup/util.py]: This is the new URL where the license is used.
Not that the old `MASTERURL` above is now defunct.

File: richkit/lookup/util.py [hunk 2]
```diff
@@ -79,22 +117,42 @@ class MaxMind_ASN_DB():
...
 
-    MASTERURL = "https://geolite.maxmind.com/download/geoip/database/GeoLite2-ASN.tar.gz"
-    MASTERFILE = temp_directory + "/asn.tar.gz"
+    MASTERURL = ( # https://dev.maxmind.com/geoip/geoipupdate/#Direct_Downloads
+        "https://download.maxmind.com/app/geoip_download?"
+        "edition_id=GeoLite2-ASN&"
+        "license_key={license_key}&"
+        "suffix=tar.gz"
+    ).format(
+        license_key=os.environ['MAXMIND_LICENSE_KEY'],
```
Comment (kidmose) [inline richkit/lookup/util.py]: Like above:
> This is the new URL where the license is used.
> Not that the old `MASTERURL` above is now defunct.

Comment (kidmose) [reply richkit/lookup/util.py]: .. but for ASN instead of country.

**Comment by kidmose** [comment]
Thanks for checking in on this :)

You're right. Definetely needs better explaining now that I revisit it!

This PR fixes two errors in `lookup.{country|asn}`, where they return an empty string.
The fails (on master branch) are here:
- https://github.com/aau-network-security/richkit/runs/425212639#step:6:18
- https://github.com/aau-network-security/richkit/runs/425212639#step:6:35

The failure in `retrieve` that you mention above is indeed still occurring, but that is another unrelated problem, that is pre.existing on the master branch. This is issue #70 and is hopefully solved with PR #71. The error on `master` is here:
https://github.com/aau-network-security/richkit/runs/425212639#step:6:52

The cause of the present issue is that Maxmind has stopped offering their geolite databases for unregistered download, and the old code handled this by returning an empty string, leading to the test results linked above (The first two).

The main contribution of this PR is to instead use a URL with a license (That can be obtained for free).

I'm pretty sure `pytest` runs through all the test cases (granted that there are no fatal errors - but only failed tests, like here). Don't these six dots mean that six test cases ran and passed:
```richkit/test/test_util.py ......                                         [100%]```
https://github.com/aau-network-security/richkit/pull/55/checks?check_run_id=435903107#step:6:15
?


Does this make things clearer?

**Comment by gianmarcomennecozzi** [comment]
Hi guys, while i was reviewing i have found that there are two functions that get the ASN Code, the first one is trough `MAXMIND`  and the other one is through `VoidURL`. you can find them in the lookup/geo.py and retrieve/util.py respectively.

I have run this branch in local and everything is working, the tests as well. The failing Test (https://github.com/aau-network-security/richkit/pull/55/checks?check_run_id=435903107#step:6:15) is related to VoidURL, so since under this pull request we are dealing with MAXMIND we are not caring about it and we ll solve it in another branch.

**Comment by kidmose** [comment]
Thanks for looking into this :)

> Hi guys, while i was reviewing i have found that there are two functions that get the ASN Code, the first one is trough `MAXMIND` and the other one is through `VoidURL`. you can find them in the lookup/geo.py and retrieve/util.py respectively.
:+1: 

> I have run this branch in local and everything is working, the tests as well. The failing Test (https://github.com/aau-network-security/richkit/pull/55/checks?check_run_id=435903107#step:6:15) is related to VoidURL, so since under this pull request we are dealing with MAXMIND we are not caring about it and we ll solve it in another branch.
:+1: 
and that other branch is hopefully #70 :)

**Comment by kidmose** [comment]
@gianmarcomennecozzi I think you pushed a commit (8736fd2) to the wrong branch?  (This is #54 and you pushed a fix for #56)
Do you mind doing a new commit where you revert this?

**Comment by kidmose** [comment]
Agreed on slack that @gianmarcomennecozzi will revert commits 8736fd2 and 9314c5d

**Pull request merged**